qemu/tests/qemu_guest_agent.py

```diff
@@ -196,16 +202,20 @@ def gagent_create(self, params, vm, *args):
         if not (args and isinstance(args, tuple) and len(args) == 2):
             raise error.TestError("Got invalid arguments for guest agent")
 
-        gagent_serial_type = args[0]
         gagent_name = args[1]
 
-        if gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_VIRTIO:
-            filename = vm.get_virtio_port_filename(gagent_name)
-        elif gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_ISA:
-            filename = vm.get_serial_console_filename(gagent_name)
-        else:
-            raise guest_agent.VAgentNotSupportedError("Not supported serial"
-                                                      " type")
+        if params.get("hotplug_gagent", "no") == "no":
+            gagent_serial_type = args[0]
+            if gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_VIRTIO:
+                filename = vm.get_virtio_port_filename(gagent_name)
+            elif gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_ISA:
+                filename = vm.get_serial_console_filename(gagent_name)
+            else:
+                raise guest_agent.VAgentNotSupportedError("Not supported serial"
+                                                          " type")
+        elif params.get("hotplug_gagent", "no") == "yes":
```

Comment:
Do you need get the params "hotplug_gagent" and judge it again? Since you have the judgement in line 207.

Reply:
Here I just want to ensure "hotplug_gagent" is yes, not others.